runtime/JavaScript/src/antlr4/index.web.js

```diff
@@ -8,7 +8,6 @@ import { default as context } from './context/index.js';
 import { default as misc } from './misc/index.js';
 import { default as tree } from './tree/index.js';
 import { default as error } from './error/index.js';
-import { default as CharStreams } from './CharStreams.js';
```

Comment:
Why on earth would you want to remove this ?

Reply:
It has a dependency on `FileStream` which is node only, so the web version needs to have this removed.

Reply:
this PR won't make it if it removes functionality

Reply:
You're duplicating code. CharStreams.node.js should reuse the code from CharStreams.web.js, adding the 2 file related entry points.

Reply:
ok, I can make that change. I duplicated code because it followed the pattern of what was done for the `index.node.js` and `index.web.js` split.

If we want to keep the code base consistent, doesn't this also mean that the same should be done to the two index files to remove duplication? There is quite a hefty amount of duplication there, and in the type definitions too.

If you can accept this duplication temporarily, I think it might be better to open a separate PR if/after this PR is merged to remove duplication in all places. It'd help limit the scope of this PR, do you agree? :-)